flowmachine/flowmachine/features/subscriber/distance_counterparts.py

```diff
@@ -0,0 +1,151 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+# -*- coding: utf-8 -*-
+"""
+Calculate metrics related with distance between caller and her/his counterparts.
+"""
+
+valid_stats = {"count", "sum", "avg", "max", "min", "median", "stddev", "variance"}
+
+from ...utils.utils import parse_tables_ensuring_columns
+from ..utilities import EventsTablesUnion
+from .metaclasses import SubscriberFeature
+from ..spatial.distance_matrix import DistanceMatrix
+
+
+class DistanceCounterparts(SubscriberFeature):
+    """
+    This class returns metrics related with the distance between event
+    initiator and her/his counterparts.
+
+    It assumes that the ID column uniquely identifies the event initiator and
+    their counterparts' event. Choose only tables for which this assumption is
+    true. In some cases, asynchronous communication like SMS might not be
+    tagged with an ID that allows one to recover the counterpart event.
+
+    Distances are measured in km.
+
+    Parameters
+    ----------
+    start, stop : str
+         iso-format start and stop datetimes
+    hours : 2-tuple of floats, default 'all'
+        Restrict the analysis to only a certain set
+        of hours within each day.
+    tables: str, default 'all'.
+        The table must have a `msisdn_counterpart` column.
```

Comment:
I just realised that we allow the `subscriber_identifier` to be either `msisdn` or `imei`, but from this comment it seems that the counterpart is always identified via `msisdn`? There doesn't seem to be an easy fix given the current table structure, so this needs to be addressed separately. It probably applies to other metrics as well which are calculated from two parties.

Reply:
The main problem here is that the MNO can only really tell the MSISDN for which the call/SMS is being directed to. Unless the subscriber is also on its network, it cannot tell anything about the IMEI of the receiving phone (at least that's my hunch). This becomes even more problematic with SMS, where the MNO sometimes can't even tell the counterpart details of an SMS transaction because they are not synchronous.

Reply:
In any case, do you want me to address this issue in this PR in any particular way? For instance, by changing the docstring, etc.

Reply:
No, I think this is fine because it clearly states the expectation on the input table. (My comment above was more of a general reflection that this needs to be addressed in flowmachine, but #358 tracks this now).